token-metadata/program/src/instruction/mod.rs

```diff
@@ -515,6 +516,8 @@ pub enum MetadataInstruction {
     #[account(7, name="spl_token_program", desc="SPL Token program")]
     #[account(8, optional, name="master_edition", desc="Unallocated edition account with address as pda of ['metadata', program id, mint, 'edition']")]
     #[account(9, optional, name="authorization_rules", desc="Token Authorization Rules account")]
+    #[args(initialize_mint: bool)]
+    #[args(update_authority_as_signer: bool)]
     Create(CreateArgs),
```

Comment:
This is going to need a Shank and Solita update for optional signers so the mint can be dynamically marked as signer or not.

Reply:
I am not sure if this is possible since Shank/Solita do not know anything about the logic that would make the account a signer or not. One way to address this is to replicate the idea of the instruction builder to the typescript SDK and add a way to register a plug-in to define additional logic.

Our currents test use a higher-level API built on top of the autogenerated one that includes this logic.

Reply:
@blockiosaurus Loris is working on a new version of solita (a.k.a. kinobi) which will support optional signers.